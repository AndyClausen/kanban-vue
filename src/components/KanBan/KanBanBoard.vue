<template>
  <v-container>
    <v-card class="mx-auto" max-width="100%">
      <v-card-title>Kanban Board</v-card-title>
      <v-card-text>
        <v-container>
          <v-row>
            <v-col v-for="column in columns" :key="column.id" cols="auto">
              <KanBanColumn :column="column" />
            </v-col>
          </v-row>
        </v-container>
      </v-card-text>
    </v-card>
  </v-container>
</template>

<script setup lang="ts">
import { useAppStore } from "@/stores/app";

const store = useAppStore();
const columns = store.columns;
</script>
