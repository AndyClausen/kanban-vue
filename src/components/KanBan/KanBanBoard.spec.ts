import { mount } from "@vue/test-utils";
import { describe, it, expect, vi } from "vitest";
import { createTestingPinia } from "@pinia/testing";
import KanBanBoard from "./KanBanBoard.vue";

const wrapper = mount(KanBanBoard, {
  global: {
    plugins: [createTestingPinia({ createSpy: vi.fn })],
  },
});

describe("KanBanBoard", () => {
  it("renders the component", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("has an add column button", () => {
    expect(wrapper.find("v-container.cursor-pointer").exists()).toBe(true);
  });
  it("adds a column when button is clicked", async () => {
    expect(wrapper.findAll("KanBanColumn").length).toBe(3);
    await wrapper.get("v-container.cursor-pointer").trigger("click");
    expect(wrapper.findAll("KanBanColumn").length).toBe(4);
  });
});
