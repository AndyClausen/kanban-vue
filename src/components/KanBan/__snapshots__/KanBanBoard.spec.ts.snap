// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`KanBanBoard > renders the component 1`] = `
"<v-container>
  <v-card color="#F2B8B4" height="80dvh" class="d-flex flex-column">
    <v-card-title class="font-weight-bold">Kanban Board</v-card-title>
    <v-card-text class="flex-grow-1 pb-0">
      <v-row class="overflow-x-scroll flex-nowrap fill-height">
        <v-col cols="auto">
          <kanbancolumn column="[object Object]"></kanbancolumn>
        </v-col>
        <v-col cols="auto">
          <kanbancolumn column="[object Object]"></kanbancolumn>
        </v-col>
        <v-col cols="auto">
          <kanbancolumn column="[object Object]"></kanbancolumn>
        </v-col>
        <v-col cols="auto">
          <v-container class="h-50 border-lg rounded-lg border-success d-flex justify-center align-center cursor-pointer" width="240px"><span class="text-success font-weight-bold text-h2">+</span></v-container>
        </v-col>
      </v-row>
    </v-card-text>
  </v-card>
</v-container>"
`;
